pkg/agent/util/syscall/syscall_windows.go

```diff
@@ -0,0 +1,367 @@
+	}
+	srcNet := p.IPNet()
+	_, srcBits := srcNet.Mask.Size()
+	if srcBits != dstBits {
+		return false
+	}
+	return srcNet.Contains(dstIPnet.IP) && dstIPnet.Contains(srcNet.IP)
+}
+
+func (p *AddressPrefix) String() string {
+	return p.IPNet().String()
+}
+
+func NewAddressPrefixFromIPNet(ipnet *net.IPNet) *AddressPrefix {
+	if ipnet == nil {
+		return nil
+	}
+	sockAddr := NewRawSockAddrInetFromIP(ipnet.IP)
+	prefixLength, _ := ipnet.Mask.Size()
+	return &AddressPrefix{
+		Prefix:       *sockAddr,
+		prefixLength: uint8(prefixLength),
+	}
+}
+
+type NlRouteProtocol uint32
+
+const (
+	RouteProtocolOther   NlRouteProtocol = 1
+	RouteProtocolLocal   NlRouteProtocol = 2
+	RouteProtocolNetMgmt NlRouteProtocol = 3
+	RouteProtocolIcmp    NlRouteProtocol = 4
+	RouteProtocolEgp     NlRouteProtocol = 5
+	RouteProtocolGgp     NlRouteProtocol = 6
+	RouteProtocolHello   NlRouteProtocol = 7
+	RouteProtocolRip     NlRouteProtocol = 8
+	RouteProtocolIsIs    NlRouteProtocol = 9
+	RouteProtocolEsIs    NlRouteProtocol = 10
+	RouteProtocolCisco   NlRouteProtocol = 11
+	RouteProtocolBbn     NlRouteProtocol = 12
+	RouteProtocolOspf    NlRouteProtocol = 13
+	RouteProtocolBgp     NlRouteProtocol = 14
+	RouteProtocolIdpr    NlRouteProtocol = 15
+	RouteProtocolEigrp   NlRouteProtocol = 16
+	RouteProtocolDvmrp   NlRouteProtocol = 17
+	RouteProtocolRpl     NlRouteProtocol = 18
+	RouteProtocolDhcp    NlRouteProtocol = 19
+
+	//
+	// Windows-specific definitions.
+	//
+	NT_AUTOSTATIC     NlRouteProtocol = 10002
+	NT_STATIC         NlRouteProtocol = 10006
+	NT_STATIC_NON_DOD NlRouteProtocol = 10007
+)
+
+type NlRouteOrigin uint32
+
+const (
+	NlroManual              NlRouteOrigin = 0
+	NlroWellKnown           NlRouteOrigin = 1
+	NlroDHCP                NlRouteOrigin = 2
+	NlroRouterAdvertisement NlRouteOrigin = 3
+	Nlro6to4                NlRouteOrigin = 4
+)
+
+type MibIpForwardRow struct {
+	Luid              uint64
+	Index             uint32
+	DestinationPrefix AddressPrefix
+	NextHop           RawSockAddrInet
+
+	SitePrefixLength  uint8
+	ValidLifetime     uint32
+	PreferredLifetime uint32
+	Metric            uint32
+	Protocol          NlRouteProtocol
+
+	Loopback             bool
+	AutoconfigureAddress bool
+	Publish              bool
+	Immortal             bool
+
+	Age    uint32
+	Origin NlRouteOrigin
+}
+
+type MibIpForwardTable struct {
+	NumEntries uint32
+	Table      [1]MibIpForwardRow
+}
+
+var (
+	modiphlpapi = syscall.NewLazyDLL("iphlpapi.dll")
+
+	procGetIpInterfaceEntry  = modiphlpapi.NewProc("GetIpInterfaceEntry")
+	procSetIpInterfaceEntry  = modiphlpapi.NewProc("SetIpInterfaceEntry")
+	procCreateIpForwardEntry = modiphlpapi.NewProc("CreateIpForwardEntry2")
+	procDeleteIpForwardEntry = modiphlpapi.NewProc("DeleteIpForwardEntry2")
+	procGetIpForwardTable    = modiphlpapi.NewProc("GetIpForwardTable2")
+	procFreeMibTable         = modiphlpapi.NewProc("FreeMibTable")
+)
+
+func GetIpInterfaceEntry(ipInterfaceRow *MibIpInterfaceRow) (errcode error) {
```

Comment:
Since there are a numbr of functions related to the objects. Abstracting these functions with an interface should make it easier to mock and test:
```
type NetIOInterface interface {
	GetIpInterfaceEntry(ipInterfaceRow *MibIpInterfaceRow) (errcode error)

	SetIpInterfaceEntry(ipInterfaceRow *MibIpInterfaceRow) (errcode error)
}

type netIO struct {
	syscallN func(trap uintptr, args ...uintptr) (r1, r2 uintptr, err syscall.Errno)
}

func NewNetIO() NetIOInterface {
	return &netIO{syscallN: syscall.SyscallN}	
}

func (n *netIO) GetIpInterfaceEntry(ipInterfaceRow *MibIpInterfaceRow) (errcode error) {
	r0, _, _ := n.syscallN(procGetIpInterfaceEntry.Addr(), uintptr(unsafe.Pointer(ipInterfaceRow)))
	if r0 != 0 {
		errcode = syscall.Errno(r0)
	}
	return
}
```